src/encdatavault_fmt_plug.c

```diff
@@ -266,51 +283,78 @@ static int valid(char *ciphertext, struct fmt_main *self)
 err:
 	MEM_FREE(keeptr);
 	return 0;
+
+}
+
+static int valid_md5(char *ciphertext, struct fmt_main *self) {
+	return valid_common(ciphertext, self, 0);
+}
+
+static int valid_pbkdf2(char *ciphertext, struct fmt_main *self) {
+	return valid_common(ciphertext, self, 1);
 }
 
-static void *get_salt(char *ciphertext)
+static void get_salt_common(char *ciphertext, struct salt *cs, int is_pbkdf2)
 {
-	static struct salt cs;
 	int i;
 	char *p = ciphertext, *ctcopy, *keeptr;
 
-	memset(&cs, 0, sizeof(cs));
```

Comment:
Please keep `cs` and its `memset` inside the function, and return a pointer to the `struct` like you did before. No need to use a separate return buffer for the two `get_salt` functions. Also, even if you do use separate return buffers, you can keep the `memset` shared in the common function.